sound/soc/sof/ipc4-topology.c

```diff
@@ -204,6 +213,22 @@ static int sof_ipc4_setup_process_set_cdata(struct snd_sof_widget *swidget)
 		process->cdata = control_data;
 		break;
 	}
+
+	module_lib_id = sof_ipc4_get_module_lib_id(swidget->module_info);
+	/* module_lib_id > 0 if module is from external library */
+	if (module_lib_id && process->cdata) {
+		struct snd_soc_component *scomp = swidget->scomp;
+		struct sof_ipc4_base_module_cfg_ext *base_ext;
+		struct sof_ipc4_control_data *control_data;
+
+		control_data = process->cdata;
+		base_ext = (struct sof_ipc4_base_module_cfg_ext *)control_data->data->data;
+		if (!base_ext->num_sink_pin_fmts || !base_ext->num_source_pin_fmts) {
```

Comment:
this is not "at least one sink pin and one source pin".

it's "one or more pins".